android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/notifications/ScopedNotificationsEmitter.java

```diff
@@ -0,0 +1,44 @@
+package versioned.host.exp.exponent.modules.universal.notifications;
+
+import android.content.Context;
+import android.util.Log;
+
+import expo.modules.notifications.notifications.emitting.NotificationsEmitter;
+import expo.modules.notifications.notifications.model.Notification;
+import expo.modules.notifications.notifications.model.NotificationRequest;
+import expo.modules.notifications.notifications.model.NotificationResponse;
+import host.exp.exponent.kernel.ExperienceId;
+
+public class ScopedNotificationsEmitter extends NotificationsEmitter {
+  private ExperienceId mExperienceId;
+
+  public ScopedNotificationsEmitter(Context context, ExperienceId experienceId) {
+    super(context);
+    mExperienceId = experienceId;
+  }
+
+  @Override
+  public void onNotificationReceived(Notification notification) {
+    if (shouldHandleNotification(notification)) {
+      super.onNotificationReceived(notification);
+    }
+  }
+
+  @Override
+  public void onNotificationResponseReceived(NotificationResponse response) {
+    if (shouldHandleNotification(response.getNotification())) {
+      super.onNotificationResponseReceived(response);
+    }
+  }
+
+  private boolean shouldHandleNotification(Notification notification) {
+    NotificationRequest notificationRequest = notification.getNotificationRequest();
+    if (notificationRequest instanceof ScopedNotificationRequest) {
+      ScopedNotificationRequest scopedNotificationRequest = (ScopedNotificationRequest) notificationRequest;
+      return scopedNotificationRequest.checkIfBelongsToExperience(mExperienceId);
+    }
+
+    Log.w("expo-notifications", "The notification's requester should be scoped.", null);
+    return false;
```

Comment:
I was wondering… if the notification requester is not scoped this is a problem… in our code (we should use the scoped requester but somehow we didn't). If that's our problem… maybe we should not reject the notification and pass it everywhere? 🤔 What do you think?